packages/node-core/src/indexer/storeModelProvider/global/global.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020-2025 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import assert from 'assert';
+import {getLogger} from '@subql/node-core/logger';
+import {Op, QueryTypes, Sequelize, Transaction} from '@subql/x-sequelize';
+import {
+  generateRewindTimestampKey,
+  GlobalData,
+  GlobalDataKeys,
+  GlobalDataRepo,
+  RewindLockInfo,
+  RewindLockKey,
+  RewindTimestampKey,
+  RewindTimestampKeyPrefix,
+} from '../../entities';
+
+export interface IGlobalData {
+  getGlobalRewindStatus(): Promise<{
+    rewindTimestamp: GlobalDataKeys[RewindTimestampKey];
+    rewindLock?: GlobalDataKeys[typeof RewindLockKey];
+  }>;
+
+  setGlobalRewindLock(rewindTimestamp: number): Promise<{needRewind: boolean}>;
+  /**
+   * Check if the height is consistent before unlocking.
+   * @param tx
+   * @param rewindTimestamp The timestamp to roll back to, in milliseconds.
+   * @returns the number of remaining rewind chains
+   */
+  releaseChainRewindLock(tx: Transaction, rewindTimestamp: number): Promise<number>;
+}
+
+const logger = getLogger('PlainGlobalModel');
+
+export class PlainGlobalModel implements IGlobalData {
+  constructor(
+    private readonly dbSchema: string,
+    private readonly chainId: string,
+    private readonly model: GlobalDataRepo
+  ) {}
+
+  private get sequelize(): Sequelize {
+    const sequelize = this.model.sequelize;
+
+    if (!sequelize) {
+      throw new Error(`Sequelize is not available on ${this.model.name}`);
+    }
+
+    return sequelize;
+  }
+
+  /**
+   * Serialize the rewind lock
+   * @param rewindTimestamp ms
+   * @param chainTotal The total number of registered chains.
+   * @returns
+   */
+  private serializeRewindLock(rewindTimestamp: number, chainTotal: number): string {
+    return JSON.stringify({timestamp: rewindTimestamp, chainNum: chainTotal});
+  }
+
+  async getGlobalRewindStatus() {
+    const rewindTimestampKey = generateRewindTimestampKey(this.chainId);
+
+    const records = await this.model.findAll({
+      where: {key: {[Op.in]: [rewindTimestampKey, RewindLockKey]}},
+    });
+    const rewindLockInfo: GlobalData<typeof RewindLockKey> | undefined = records
+      .find((r) => r.key === RewindLockKey)
+      ?.toJSON();
+    const rewindTimestampInfo: GlobalData<RewindTimestampKey> | undefined = records
+      .find((r) => r.key === rewindTimestampKey)
+      ?.toJSON();
```

Comment:
I think you could do `toJSON` on `records` then use a destructor.
Also `toJSON` is generic so you can pass the type in there.